presto-testing/src/main/java/io/prestosql/testing/AbstractTestIntegrationSmokeTest.java

```diff
@@ -372,4 +412,27 @@ public void testSelectInformationSchemaColumns()
         assertQuerySucceeds("SELECT * FROM information_schema.columns WHERE table_catalog = '" + catalog + "' AND table_name LIKE '%'");
         assertQuery("SELECT column_name FROM information_schema.columns WHERE table_catalog = 'something_else'", "SELECT '' WHERE false");
     }
+
+    protected void assertPushedDown(String sql)
+    {
+        assertPushedDown(sql, sql);
+    }
+
+    protected void assertPushedDown(String actual, String expectedOnH2)
+    {
+        assertQuery(actual, expectedOnH2);
+
+        transaction(getQueryRunner().getTransactionManager(), getQueryRunner().getAccessControl())
+                .execute(getSession(), session -> {
+                    Plan plan = getQueryRunner().createPlan(session, actual, WarningCollector.NOOP);
+                    assertPlan(
+                            session,
+                            getQueryRunner().getMetadata(),
+                            (node, sourceStats, lookup, ignore, types) -> PlanNodeStatsEstimate.unknown(),
+                            plan,
+                            PlanMatchPattern.output(
+                                    PlanMatchPattern.exchange(
+                                            PlanMatchPattern.node(TableScanNode.class))));
```

Comment:
don't you need to assert here what was pushed down?

Reply:
i assert that "everything" got pushdown and "yet" results are correct.